emails/views.py

```diff
@@ -118,6 +123,159 @@ def _index_DELETE(request_data, user_profile):
     return redirect('profile')
 
 
+@csrf_exempt
+def sns_inbound(request):
+    # We can check for some invalid values in headers before processing body
+    topic_arn = request.headers.get('X-Amz-Sns-Topic-Arn', None)
+    if not topic_arn:
+        logger.error('SNS inbound request without X-Amz-Sns-Topic-Arn')
+        return HttpResponse(
+            'Received SNS request without Topic ARN.', status=400
+        )
+    if topic_arn != settings.AWS_SNS_TOPIC:
+        logger.error(
+            'SNS message for wrong ARN',
+            extra={
+                'configured_arn': settings.AWS_SNS_TOPIC,
+                'received_arn': topic_arn,
+            }
+        )
+        return HttpResponse(
+            'Received SNS message for wrong topic.', status=400
+        )
+
+    message_type = request.headers.get('X-Amz-Sns-Message-Type', None)
+    if not message_type:
+        logger.error('SNS inbound request without X-Amz-Sns-Message-Type')
+        return HttpResponse(
+            'Received SNS request without Message Type.', status=400
+        )
+    if message_type not in SUPPORTED_SNS_TYPES:
+        logger.error(
+            'SNS message for unsupported type',
+            extra={
+                'supported_sns_types': SUPPORTED_SNS_TYPES,
+                'message_type': message_type,
+            }
+        )
+        return HttpResponse(
+            'Received SNS message for unsupported Type: %s' % message_type,
+            status=400
+        )
```

Comment:
I recommend creating a function `validate_sns_header(topic_arn, message_type)` which does this check.

Reply:
Gonna merge early to test dev env. Can you file an issue for this?

Reply:
Filted #364 for this.